### DISCUSSION TITLE
Making the core protocol Disability aware

### DISCUSSION THREAD

**Discussion by muneender-kp** [beckn/protocol-specifications#382] [Core Specification]
This initiative aims to improve accessibility by accommodating users with disabilities through user-friendly interfaces and assistive technology compatibility, broadening the protocol's reach and impact.

Inviting your thoughts

**Comment by Anshadixit** [comment]
Nice initiative.
I see use-cases can be spread around both sides-

Buyer with disabilities need to convey their special needs while looking for an option.
A seller with special helps/ arrangements should be able to publish its catalog to all types of BAPs.

**Reply by shantiraghavan** [reply]
exactly -  I think there should be a way in the protocol to convey the information whether-
•	A person has special needs (can't take calls, can't see instructions, etc.)
•	Or a item is offered with special features (accessible clothing, accessible restaurants, trained driver ofcab to handle wheel chair, etc.)
What can be the best way to make sure every provider/ consumer pair can communicate these needs and availability?

**Reply by kshitijahuja** [reply]
This specification also needs have integration with the service provider's systems otherwise their staff will continue to ignore these specific needs. Take for instance the banks. They do not appear to have a feature to flag deaf customers, the outcome of which is, every time you try to reach them through email or Twitter, you are incessantly bombed with callbacks. 

Consider adding a field to mark whether the service provider received this badge/flag specifications back into their system or not. And at the same time, encourage them to update their systems to have this feature built. (They might just have it around but not being used).

**Reply by shantiraghavan** [reply]
Excellent point Kshitij! Ur point will be very helpful in building the beckn layer

**Reply by shantiraghavan** [reply]
Excellent point Kshitij! Ur point will be very helpful in building the beckn layer

**Comment by muneender-kp** [comment]
There's a young lady, who I know, is visually impaired. She relies on the Namma Yatri app, made using the ONDC/Beckn Protocol, to get around the city. In the past few months, she's spent over 1 Lakh rupees on rides in last few months. 

Now, here's the exciting part: this isn't just about her. It's a chance for app creators using this protocol to reach more people and earn more money. It's like a hidden treasure waiting to be discovered! At protocol level, How do we empower builders  to tap into this market ?

**Reply by shantiraghavan** [reply]
you are right!
my brother who is blind lives with my mom and is the **bread winner** for the family. He works in an asia pacific role in a large multi national. He is the one who pays the bills, books travel for my mom, uses mutual funds, uses multiple banking apps, goes out for entertainment and restaurants, has a great friend circle.. 
He finds most of the digital apps inaccessible with his screen reader (software "reads out" what he needs to see). He shops with the "less inaccessible" of the lot! If only these app developers could know about people like my brother and build for them and attract a **new consumer base**

**Comment by animaagrawal** [comment]
this is something amazing

**Reply by shantiraghavan** [reply]
Thanks anima.

**Comment by kshitijahuja** [comment]
Identification object for PwDs needs to be included, if not already included. Customer, Person, Contact objects do not seem to have any specific field for marking the PwDs. I feel, the following fields could be included:

- Whether the individual is a Person with Disability [boolean, true/false]
- Type of disability [dropdown, string of shortcodes] 
- Extent of disability [floating point number]
- Preferred mode of communication [string (email, phone, postal mail etc.) OR specific pointer to existing contact field, if possible]

How could this be implemented without disrupting the existing protocol?

**Reply by venkatramanm** [reply]
Tags at person level can be used right now. The tag names, and values can be decided by the domain spec community . 

It seems like a good idea to add these as first class attributes instead of tags. Would need some discussions with core protocol community to arrive at it. 

My take is first work with them domain community and once baked. We can absorb in core protocol .

**Reply by kshitijahuja** [reply]
Thanks! Makes sense to discuss with domain specific stakeholders for consensus.

WRT tags, two things come to my mind off-the-bat:
- Are there strict controls on tags available? These attributes will most likely have standardized (as these are govt-defined) key-values.
- Is search availability on Tags available, so these could be looked up easily?

**Reply by kshitijahuja** [reply]
@shantiraghavan How do you reckon we could give this a push for discussion among other stakeholders and community?

**Comment by shantiraghavan** [comment]
would love ur suggestions. But I was planning to have a series of calls with the techie pwd community. Get them to know about this as a start. After that GitHub discussions by community will take over

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by kshitijahuja** [reply]
That should be a good start. Domain expertise, rather than tech expertise is the need at this stage.
